src/libMessage/Messenger.cpp

```diff
@@ -113,6 +113,20 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
   Serializable::SetNumber<T>(dst, offset, number, S);
 }
 
+template <class K, class V>
+bool CopyWithSizeCheck(const K& arr, V& result) {
```

Comment:
Can add some unit test case for this function?

Reply:
It's a bit tricky I think to add unit test case for this as it is a local function that does not need to be exposed outside of Messenger.cpp.  For now I think the easiest way to test is to run testnet and trigger the call to this function in all the other functions (I'll do this for @robin-thomas)

Reply:
We can put this function in Messenger.h and then can add unit test for it in unit test cpp files.

Reply:
Updated as suggested